### DISCUSSION TITLE
createPullRequest API not working

### DISCUSSION THREAD

**Discussion by cjiang-ias** [community/community#24349] [Apps, API and Webhooks]
<p>Hey, I’ve added to my query everything that you indicated a need for in the <a href="https://developer.github.com/v4/input_object/createpullrequestinput/" rel="nofollow noopener">docs</a> for the GraphQL API’s createPullRequestInput, but I’m still getting the following errors:</p>
<pre><code>[data:[createPullRequest:[pullRequest:null]], errors:[[type:UNPROCESSABLE, path:[createPullRequest], locations:[[line:3, column:7]], message:Head sha can't be blank, Base sha can't be blank, No commits between master and CICD-3853_test_branch, Head ref must be a branch]]]
</code></pre>
<p>Here’s my code for reference:</p>
<pre><code>// query
    mutation createPrForArns {
      createPullRequest(input:{ baseRefName: "master", body: "my test PR body", headRefName: "my-feature-branch", repositoryId: "my-repo-node-id", title: "test" }) {
        pullRequest {
          resourcePath
        }
      }
    }
</code></pre>
<p>Any help around this would be highly appreciated.</p>
<p>P.S. It would be really helpful to have examples for the mutations in your API documentation. Thanks.</p>

**Comment by cjiang-ias** [comment]
<p>By the way, there <em>is</em> a commit between the testing branch and the base branch of master - I added a test commit and pushed it up to the testing branch. </p>

**Comment by parithiban** [comment, marked as answer]
<p><a class="mention" href="/cjiang-ias">@cjiang-ias</a> </p>
<p>This is working fine for me.</p>
<p>What is the error message that you are getting?</p>
<p>Also, is your <strong>testing branch</strong> pushed to the origin or it is a local branch?</p>
<pre><code>mutation {
	createPullRequest(input: {
		repositoryId: "X-X-X-X",
		baseRefName: "master",
		title: "Pull request from github V4",
		headRefName: "parithibang-patch-3"
	}) {
		pullRequest {
			body
			title
		}
	}
}
</code></pre>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/8/817b9dcb8a28447f459f2b34245d3162d891f359.png" data-download-href="/uploads/short-url/itsrjzvUMvzpM4iRgsJYjtf53GV.png?dl=1" title="Screen Shot 2020-02-24 at 6.49.32 PM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/7281166/181084323-70bf4649-79b3-4f5b-9274-ab989d12bbc7.png" alt="Screen Shot 2020-02-24 at 6.49.32 PM.png" data-base62-sha1="itsrjzvUMvzpM4iRgsJYjtf53GV" width="689" height="254" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/8/817b9dcb8a28447f459f2b34245d3162d891f359_2_689x254.png, https://aws1.discourse-cdn.com/github/optimized/2X/8/817b9dcb8a28447f459f2b34245d3162d891f359_2_1033x381.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/8/817b9dcb8a28447f459f2b34245d3162d891f359_2_1378x508.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/8/817b9dcb8a28447f459f2b34245d3162d891f359_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-02-24 at 6.49.32 PM.png</span><span class="informations">2558×942 263 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by cjiang-ias** [comment]
<p>Thanks for following up, <a class="mention" href="/parithiban">@parithiban</a>. Turns out it was just the wrong repo ID. The mutation indeed works. </p>

**Comment by jpruiz114** [comment]
<p>Thanks for sharing, works fine.</p>
